core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -977,6 +977,20 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
         listing.delete(classOf[LogInfo], log.logPath)
       }
     }
+    // Delete log files that don't exist in listing database and exceed the configured max age.
+    Option(fs.listStatus(new Path(logDir))).map(_.toSeq).getOrElse(Nil)
```

Comment:
Since is this an additional `fs.listStatus` call, do you think this will be expensive since we are making two list status calls - one in `checkForLogs` and one in `cleanLogs` ? Is there any way to include this logic in `checkForLogs` itself ?

Reply:
> Since is this an additional `fs.listStatus` call, do you think this will be expensive since we are making two list status calls - one in `checkForLogs` and one in `cleanLogs` ? Is there any way to include this logic in `checkForLogs` itself ?

Thanks fo your review, i will move this logic to `checkForLogs`.